tests/templates/test_layers.py

```diff
@@ -477,6 +478,43 @@ def test_random_layer_weights(self, n_subsystems, tol):
         params_flat = [item for p in params for item in p]
         assert np.allclose(weights.flatten(), params_flat, atol=tol)
 
+    def test_tensor_unwrapped_well(self, monkeypatch):
```

Comment:
If I understand this correctly, this test checks that the weight argument in `gate` in `random_layer` is always a float, even when passing a tensor object, meaning that unwrapping is taking place.

Reply:
Correct added more description